### PR TITLE
2299 fix(frontend): corrected tag spacing in settings object item table row

### LINKED ISSUE #2299: Correct tags spacings
# ❗Edited

## Current look

On `Settings/Objects` the **Standard** and **Custom** tags look like this:

image

## Correction

- The tag text should be vertically centered (and the height should be 16px)
- The Left and right padding should be 4px instead of 8px.

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #2299 

Text is now centred and horizontal padding rectified.

Screenshot 2023-10-31 at 9 34 40 PM

### CODE REVIEW

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectItemTableRow.tsx [hunk 1]
```diff
@@ -26,7 +26,11 @@ const StyledNameTableCell = styled(TableCell)`
 
 const StyledTag = styled(Tag)`
   box-sizing: border-box;
-  height: ${({ theme }) => theme.spacing(4)};
+  height: 16px;
+  padding-bottom: auto;
+  padding-left: 4px;
+  padding-right: 4px;
+  padding-top: auto;
```
Comment (thaisguigon) [inline front/src/modules/settings/data-model/object-details/components/SettingsObjectItemTableRow.tsx:33]: Hi @Kanav-Arora, thank you for contributing!

In the design system there are different sizes for the `Tag` component, which are not implemented yet: https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=3063%3A54900&mode=design&t=Oz0MjRJZUU0tq8P1-1

Could you implement a `size` prop for the `Tag` component, with possible values: `small`, `medium` and `large`, according to Figma? The default would be "large" as it is the closest to the existing one (height: 24px, padding-left and right: 8px. We can remove the padding top and bottom).
Then we would use `size="small"` in `SettingsObjectItemTableRow.tsx`.

Also, please use the `theme.spacing(...)` method in the styles (`theme.spacing(1)` is 4px, `theme.spacing(2)` is 8px, etc.).

**Review by thaisguigon — CHANGES_REQUESTED** [review]

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectItemTableRow.tsx (hunk 1 above)
Comment (Kanav-Arora) [reply front/src/modules/settings/data-model/object-details/components/SettingsObjectItemTableRow.tsx:33]: Sure
I'll do the needful

**Comment by Kanav-Arora** [comment]
Hi @thaisguigon 
I have pushed the required changes. PFA

Screenshot 2023-11-04 at 11 17 02 PM

**Comment by Kanav-Arora** [comment]
Hi @thaisguigon 
Please review the changes

**Review by thaisguigon — CHANGES_REQUESTED** [review]

File: front/src/modules/ui/display/tag/components/Tag.tsx [hunk 2]
```diff
@@ -33,25 +42,32 @@ const StyledTag = styled.h3<{
   font-style: normal;
   font-weight: ${({ theme }) => theme.font.weight.medium};
   gap: ${({ theme }) => theme.spacing(2)};
+  height: ${({ size, theme }) => theme.spacing(sizeHeight[size])};
```
[the remark this answers is not in the document]
Comment (Kanav-Arora) [reply front/src/modules/ui/display/tag/components/Tag.tsx:45]: Ok🥲

**Comment by Kanav-Arora** [comment]
Hi @thaisguigon 

I have made the requested changes, please review

Screenshot 2023-11-06 at 9 56 06 PM

**Review by thaisguigon — APPROVED** [review]
LGTM, thank you!

**Pull request merged**